pkg/resource/training_job/sdk.go

```diff
@@ -1060,6 +1060,9 @@ func (rm *resourceManager) sdkUpdate(
 	if warmpool_diff && profiler_diff {
 		return latest, errors.New("[ACK_SM] Cannot update Warm pool and Profiler at the same time.")
 	}
+	if !warmpool_diff && !profiler_diff {
```

Comment:
Added this to prevent an infinite loop, but this might be a problem if there is a server side default update/bug. Is there a better way to handle this?

Reply:
Using DifferenceExcept does not work well with unit tests as unit tests do a delta after the update as opposed to before.